src/Compilers/CSharp/Test/Syntax/Parsing/UsingDirectiveParsingTests.cs

```diff
@@ -4077,4 +4037,1224 @@ public void TestMakeref()
+
+        N(SyntaxKind.CompilationUnit);
+        {
+            N(SyntaxKind.UsingDirective);
+            {
+                N(SyntaxKind.UsingKeyword);
+                N(SyntaxKind.StaticKeyword);
+                N(SyntaxKind.UnsafeKeyword);
+                N(SyntaxKind.QualifiedName);
+                {
+                    N(SyntaxKind.QualifiedName);
+                    {
+                        N(SyntaxKind.QualifiedName);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "System");
+                            }
+                            N(SyntaxKind.DotToken);
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "Collections");
+                            }
+                        }
+                        N(SyntaxKind.DotToken);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "Generic");
+                        }
+                    }
+                    N(SyntaxKind.DotToken);
+                    N(SyntaxKind.GenericName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "List");
+                        N(SyntaxKind.TypeArgumentList);
+                        {
+                            N(SyntaxKind.LessThanToken);
+                            N(SyntaxKind.ArrayType);
+                            {
+                                N(SyntaxKind.PointerType);
+                                {
+                                    N(SyntaxKind.PredefinedType);
+                                    {
+                                        N(SyntaxKind.IntKeyword);
+                                    }
+                                    N(SyntaxKind.AsteriskToken);
+                                }
+                                N(SyntaxKind.ArrayRankSpecifier);
+                                {
+                                    N(SyntaxKind.OpenBracketToken);
+                                    N(SyntaxKind.OmittedArraySizeExpression);
+                                    {
+                                        N(SyntaxKind.OmittedArraySizeExpressionToken);
+                                    }
+                                    N(SyntaxKind.CloseBracketToken);
+                                }
+                            }
+                            N(SyntaxKind.GreaterThanToken);
+                        }
+                    }
+                }
+                N(SyntaxKind.SemicolonToken);
+            }
+            N(SyntaxKind.EndOfFileToken);
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void UsingStatic_UnsafeType_CSharp11_NoUnsafeFlag()
```

Comment:
>UsingStatic_UnsafeType_CSharp11_NoUnsafeFlag [](http://example.com/codeflow?start=16&length=44)

It looks like the added tests each have 4 variants, { C# 11, C# next } x { no -unsafe, -unsafe }, with each test tests binding and parsing separately. But the results of the parser should be the same across all variants (and the parser does not depend on `CSharpCompilationOptions` which contains the unsafe setting).

Given that, would it make sense to combine the variants? For instance:
```csharp
[Theory]
[InlineData(LanguageVersion.CSharp10)]
[InlineData(LanguageVersion.CSharp11)]
public void UsingStatic_UnsafeType(LanguageVersion languageVersion)
{
    var parseOptions = TestOptions.Regular.WithLanguageVersion(languageVersion);
    var text = "...";

    if (languageVersion == LanguageVersion.CSharp10)
    {
        CreateCompilation(text, parseOptions: parseOptions).VerifyDiagnostics(...);
        CreateCompilation(text, parseOptions: parseOptions, options: TestOptions.UnsafeDebugDll).VerifyDiagnostics(...);
    }
    else
    {
        CreateCompilation(text, parseOptions: parseOptions).VerifyDiagnostics(...);
        CreateCompilation(text, parseOptions: parseOptions, options: TestOptions.UnsafeDebugDll).VerifyDiagnostics(...);
    }

    UsingTree(text, options: parseOptions);
    N(SyntaxKind.CompilationUnit);
    {
        ...
    }
}
```

Reply:
I find theories, which then have different behavior on units, nigh unreadable. I spend more time just trying to understand what the final test is actually testing. 

It's also a huge pita when one variant of the test fails, and you continually have to revise and run all the other variants. 

I'm a big fan of a test having an input and validating it. A true unit test.